src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -84,6 +84,29 @@ bool CImageLoaderTGA::isALoadableFileFormat(io::IReadFile* _file) const
 	return true;
 }
 
+// convertColorFlip() does color conversion as well as taking care of properly flipping the given image.
+template <typename T, E_FORMAT srcFormat, E_FORMAT destFormat>
+static void convertColorFlip(const T *src, T *dest, core::vector3d<uint32_t> &size, int c, bool flip)
```

Comment:
this is the kind of thing that would be **very** useful in top level header, maybe in `./include/irr/video/convertColor.h`